lxd/events.go

```diff
@@ -35,7 +35,31 @@ func (r *eventsServe) String() string {
 }
 
 func eventsSocket(d *Daemon, r *http.Request, w http.ResponseWriter) error {
+	allProjects := queryParam(r, "all-projects") == "true"
+	projectQueryParam := queryParam(r, "project")
+	if allProjects && projectQueryParam != "" {
+		response.BadRequest(fmt.Errorf("'project' cannot be set if 'all-projects' is set"))
+		return nil
+	}
+
+	if !allProjects && projectQueryParam != "" {
+		_, err := d.cluster.GetProject(projectQueryParam)
+		switch err {
+		case nil:
+			break
+		case db.ErrNoSuchObject:
+			response.BadRequest(fmt.Errorf("project '%s' does not exist", projectQueryParam)).Render(w)
```

Comment:
Should be "Project %q not found" - so we have an upper case first letter and using double quota `%q` for quoting string params. And we tend to use "not found" these days (although I'm sure there are some "does not exist" ones left over still) :)